near_global_shallow_water_quarter_degree.jl

```diff
@@ -0,0 +1,222 @@
+                                              longitude = (-180, 180),
+                                              latitude = latitude,
+                                              z = (0, 0),
+                                              halo = (4, 4),
+                                              topology = (Periodic, Bounded, Flat),
+                                              precompute_metrics = true)
+
+grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBoundary(boundary))
+
+τˣ = arch_array(arch, τˣ)
+τʸ = arch_array(arch, τʸ)
+    
+#####
+##### Boundary conditions / time-dependent fluxes 
+#####
+
+@inline current_time_index(time, tot_months)     = mod(unsafe_trunc(Int32, time / thirty_days), tot_months) + 1
+@inline next_time_index(time, tot_months)        = mod(unsafe_trunc(Int32, time / thirty_days) + 1, tot_months) + 1
+@inline cyclic_interpolate(u₁::Number, u₂, time) = u₁ + mod(time / thirty_days, 1) * (u₂ - u₁)
+
+@inline function boundary_stress_u(i, j, k, grid, clock, fields, p)
+    time = clock.time
+    n₁ = current_time_index(time, Nmonths)
+    n₂ = next_time_index(time, Nmonths)
+
+    @inbounds begin
+        τ₁ = p.τ[i, j, n₁]
+        τ₂ = p.τ[i, j, n₂]
+    end
+
+    if fields.h[i, j, k] > 0 
+        return (cyclic_interpolate(τ₁, τ₂, time) - p.μ * fields.u[i, j, k]) / p.hₘ
+    else
+        return 0.0
+    end
+end
+
+
+@inline function boundary_stress_v(i, j, k, grid, clock, fields, p)
+    time = clock.time
+    n₁ = current_time_index(time, Nmonths)
+    n₂ = next_time_index(time, Nmonths)
+
+    @inbounds begin
+        τ₁ = p.τ[i, j, n₁]
+        τ₂ = p.τ[i, j, n₂]
+    end
+    if fields.h[i, j, k] > 0 
+        return (cyclic_interpolate(τ₁, τ₂, time) - p.μ * fields.v[i, j, k]) / p.hₘ
+    else
+        return 0.0
+    end
+end
+
+# Linear bottom drag:
+μ = 0.001 # ms⁻¹
+
+Fu = Forcing(boundary_stress_u, discrete_form = true, parameters = (μ = μ, τ = τˣ, hₘ = hₘ))
+Fv = Forcing(boundary_stress_v, discrete_form = true, parameters = (μ = μ, τ = τʸ, hₘ = hₘ))
+
+using Oceananigans.Models.ShallowWaterModels: VectorInvariantFormulation
+using Oceananigans.Advection: VelocityStencil
+using Oceananigans.TurbulenceClosures: HorizontalDivergenceFormulation
+
+model = ShallowWaterModel(grid = grid,
+			              gravitational_acceleration = 9.8055,
+                          momentum_advection = WENO5(vector_invariant = VelocityStencil()),
+                          coriolis = HydrostaticSphericalCoriolis(),
+                          forcing = (u=Fu, v=Fv),
+            			  bathymetry = bat,
+                          closure = ScalarDiffusivity(HorizontalDivergenceFormulation(), ν = 100),
+			              formulation = VectorInvariantFormulation())
+
+#####
+##### Initial condition:
+#####
+
+h_init = bat
+set!(model, h=h_init)
+@info "model initialized"
+
+#####
+##### Simulation setup
+#####
+
+Δt = 10seconds #1minutes  # for initialization, then we can go up to 6 minutes?
```

Comment:
This comment suggests that the short time is needed because of the initial adjustmenet but then we can use a larger time step.  That would be great. I don't suppose the time step wizard will do this as it only deals with advection and diffusion, right?